src/engine/sim_engine/engine.rs

```diff
@@ -40,7 +47,50 @@ impl Engine for SimEngine {
         let redundancy = calculate_redundancy!(redundancy);
 
         match self.pools.get_by_name(name) {
-            Some(_) => Ok(CreateAction::Identity),
+            Some((_, pool)) => {
+                let input_devices = blockdev_paths
+                    .iter()
+                    .cloned()
+                    .collect::<HashSet<&Path, RandomState>>();
```

Comment:
We've always let compiler infer the random state in other contexts, maybe that is possible here.

Reply:
Okay, it existed as an explicit declaration previously in the sim engine so I'll switch over to inference where I find it and see what happens.

Reply:
One of the `HashSet`s is unable to have `RandomState` removed. The compiler can't infer that type parameter. I'm leaving that one as is.

Reply:
Resolved.